src/overcooked_ai_py/mdp/overcooked_env.py

```diff
@@ -252,7 +252,8 @@ def reset(self, regen_mdp=True, outside_info={}):
         if regen_mdp:
             self.mdp = self.mdp_generator_fn(outside_info)
             self._mlam = None
-            self._mp = None
+            # The motion planner (self.mp) will almost certainly be used. Initialize here to avoid pickle loading issues
+            self.mp
```

Comment:
Do you not still need the `self._mp = None` line here in order to cause the `self.mp` line to recompute the motion planner? My understanding here is that it'll still return the old one since it was never invalidated

Reply:
Also this might be too complicated for it to be worth it, but would it be possible to have `self._mp` point at `self._mlam.motion_planner.motion_planner`? The way it's set up now I believe we are redundantly computing two separate copies of the same `MotionPlanner`

Reply:
Thanks for pointing out the issue! It has been addressed by restoring `self._mp =None`, but upon calling `self.mp`, the `env` will first look into `self.mlam.motion_planner`